src/components/gui/is-renderer-supported.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import Renderer from 'scratch-render';
+
+import {
+    SVGRenderer as V2SVGAdapter,
+    BitmapAdapter as V2BitmapAdapter
+} from 'scratch-svg-renderer';
+
+let _isRendererSupported = null;
+
+const isRendererSupported = WrappedComponent => (
+    function IsRendererSupported (props) {
+        const {vm} = props;
+        if (_isRendererSupported === null) {
+            if (vm.renderer) {
+                _isRendererSupported = true;
+            } else {
+                const canvas = document.createElement('canvas');
+                _isRendererSupported = Renderer.isSupported(canvas);
+                if (_isRendererSupported) {
+                    // Sharing the canvas that is used to check if renderer is
```

Comment:
This makes a lot of sense, but it is definitely located in a surprising place. That is, the renderers canvas is created in a file called `is-renderer-supported`. Perhaps caching the test canvas is something the renderer should do? Or we make the Renderer constructor throw if it isn't supported, moving `isSupported` inside the constructor? 

Additionally, is there a different place/way to organize the attachment of vm "extras" like the renderer, image adapters and (not shown here) the audio engine, storage system. It always has to be done delicately to avoid re-attaching, and it makes sense that they wouldn't be attached unless there was a stage component being shown, but it still seems bad to have them attached in `is-renderer-supported`.